tests/unit/test_cv_device_tools.py

```diff
@@ -0,0 +1,164 @@
+
+@pytest.fixture
+def setup(apply_mock, mock_cvpClient):
+    """
+    setup - setup method to apply mocks and patches
+    """
+    mock_ansible_module, mock__get_device = apply_mock(MOCK_LIST)
+    mock_ansible_module.fail_json.side_effect = mock_cvpClient.api.fail_json
+    cv_tools = CvDeviceTools(mock_cvpClient, mock_ansible_module)
+
+    return mock_ansible_module, mock__get_device, cv_tools
+
+@pytest.mark.state
+class TestDecommissionDevice():
+    """
+    Contains unit tests for decommission_device()
+    """
+
+    @pytest.mark.parametrize(
+        "device_data, expected, expected_fail_json_call_msg",
+        [
+            (device_data, True, ""),
+            (device_data_invalid, False, "Device decommissioning failed due to Device does "
+                                         "not exist or is not registered to decommission"),
+        ],
+    )
+    def test_decommission_device(self, setup, device_data, expected, expected_fail_json_call_msg):
+        """
+        Tests decommission_device() method for state_absent
+        if device_data['serialNumber'] is correct:
+            expected = true
+        else:
+            expected = false and error_msg
+        """
+        user_topology = DeviceInventory(data=device_data)
+        mock_ansible_module, mock__get_device, cv_tools = setup
+        mock__get_device.return_value = device_data[0]  # mocked for get_device_facts, device_data is in tests/datadevice_tools_unit.py
+        if expected:
+            result = cv_tools.decommission_device(user_inventory=user_topology)
+            assert result[0].success == expected
+            assert result[0].changed == expected
+        else:
+            with pytest.raises(SystemExit) as pytest_error:
+                _ = cv_tools.decommission_device(user_inventory=user_topology)
+            assert pytest_error.value.code == 1
+            expected_call = [call.fail_json(msg=expected_fail_json_call_msg)]
+            assert mock_ansible_module.mock_calls == expected_call
+
+@pytest.mark.state
+class TestResetDevice():
+    """
+    Contains unit tests for reset_device()
+    """
+    @pytest.mark.parametrize(
+        "device_data, expected",
+        [
+            (device_data, True),
+            (device_data_invalid, False),
+        ],
+    )
+    def test_reset_device(self, setup, device_data, expected):
+        """
+        Tests reset_device method for state factory_reset
+        device_data: dummy_device_data
+        if device_data['parentContainerName'] is "undefined":
+            expected = false
+        else:
+            expected = true
+        """
+        user_topology = DeviceInventory(data=device_data)
+        _, _, cv_tools = setup
+        result = cv_tools.reset_device(user_inventory=user_topology)
+        assert result[0].success == expected
+        assert result[0].changed == expected
+        if expected:
+            assert result[0].taskIds == ['57']
```

Comment:
Not clear what this '57' is here - probably give it as a parameter to the test, explain what it is (task id?) and then if it is present then test for the value being the expected one.

Explain what it is in the docstring - what are we checking here?

Reply:
Updated it